pkg/agent/runner.go

```diff
@@ -778,7 +779,12 @@ func (r *Runner) DoSchedulerRequest(
 	}
 	request.Header.Set("content-type", "application/json")
 
-	logger.Info("Sending request to scheduler", zap.Any("request", reqData))
+	if reqData.LastPermit != nil && *reqData.LastPermit == reqData.Resources {
+		// If the last permit is the same as the current request, we can skip request logging.
+		logger.Debug("Sending request to scheduler", zap.Any("request", reqData))
+	} else {
+		logger.Info("Sending request to scheduler", zap.Any("request", reqData))
+	}
```

Comment:
I'd rather keep this as debug-only and put the conditional in `exec_plugin.go`, so that it takes place while we hold the lock (just nicer to prioritize log lines that are guaranteed to be in the same order as the operations that took place :sweat_smile:)

Reply:
My idea was that `exec_plugin.go` doesn't know exact `reqBody` sent to the scheduler, so request log in `runner.go` has more context.

Also I'm not sure about the conditional, what is the best balance between logs cost / usefulness here in your opinion?

WDYT about such log levels and no conditions?
1. [info] Starting plugin request (exec_plugin.go)
2. [debug] Sending request to scheduler (runner.go)
3. [debug] Received response from scheduler (runner.go)
4. [info] Plugin request successful (exec_plugin.go)

Reply:
> WDYT about such log levels and no conditions?

That makes sense, yeah - I think we _could_ also change (1) to debug in the cases where it doesn't change, but it's easy enough to change that in a follow-up

> My idea was that exec_plugin.go doesn't know exact `reqBody` sent to the scheduler, so request log in `runner.go` has more context.

Yeah, that's what I was thinking first as well, but eventually thought it'd be better to keep the guarantees around log lines being in the right order -- especially because IIRC `reqBody` can be exactly determined from the action given to `exec_plugin.go`